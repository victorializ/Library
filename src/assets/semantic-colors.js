export const colors = {
    primary: 'olive', 
    secondary: 'black',
    grey: 'grey',
    red: 'red'
}