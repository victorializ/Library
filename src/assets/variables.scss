$space-sm: 30px;
$space-lg: 70px;

@mixin centered-vertical-container($justify: 'center', $align: 'center') {
    display: flex;
    justify-content: $justify;
    align-items: $align;
    flex-direction: column;
}

@mixin centered-horizontal-container {
    display: flex;
    justify-content: center;
    align-items: center;
}
