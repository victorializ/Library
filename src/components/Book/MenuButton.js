import React from 'react';

import { Button } from 'semantic-ui-react'

import { constants } from '../../i18n';
import { colors } from '../../assets/semantic-colors';

const MenuButton = ({name, selectedButton, clickHandler}) => {

  const getButtonColor = (buttonName, selected) =>  {
    return buttonName === selected ? colors.primary : colors.grey;
  }

  return (
    <Button 
      onClick={() => clickHandler(name)}
      color={getButtonColor(name, selectedButton)}>
        {constants[name]}
    </Button>
  )
};

export { MenuButton };