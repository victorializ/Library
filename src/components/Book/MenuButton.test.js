import React from 'react';
import { shallow } from 'enzyme';
import { colors } from '../../assets/semantic-colors';

import { MenuButton } from './MenuButton';

describe('MenuButton component', () => {
    it('renders books list component correctly', () => {
        expect(shallow(<MenuButton />)).toMatchSnapshot();
    });
    it('renders selected button', () => {
        const wrapper = shallow(<MenuButton name='B' selectedButton='B' />);
        expect(wrapper.find('.menu__button').prop('color')).toEqual(colors.primary);
    });
    it('renders unselected button', () => {
        const wrapper = shallow(<MenuButton name='B' selectedButton='A' />);
        expect(wrapper.find('.menu__button').prop('color')).toEqual(colors.grey);
    });
    it('handles click event', () => {
        const handleClick = jest.fn();
        const name = 'B';
        const wrapper = shallow(<MenuButton name={name} clickHandler={handleClick}/>);
        wrapper.find('.menu__button').simulate('click');
        expect(handleClick.mock.calls[0][0]).toEqual(name);
    });
});