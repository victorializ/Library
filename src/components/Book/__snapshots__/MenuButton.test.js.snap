// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton component renders books list component correctly 1`] = `
<Button
  as="button"
  className="menu__button"
  color="olive"
  onClick={[Function]}
/>
`;
